docs/config.rst

```diff
@@ -218,6 +218,21 @@ Defaults for each setting in this section are looked up in the::
 
 Complete list of settings that you can put into ``testenv*`` sections:
 
+.. conf:: host_env ^ true|false ^ false
+
+    If ``true`` tox will not create a virtual environment to run your commands inside and instead use the hosts
+    tox environment (the python from where tox is running). Some options for such environments cannot be changed,
+    and setting these config values will be ignored:
+
+    - :conf:`skip_install` is always ``True``,
+    - :conf:`deps` is always empty (in essence no install operation allowed to not break the host tox),
+    - :conf:`basepython` will always be the full path of the host python interpreter.
+
+    If one does specify pip install via the :conf:`commands` we'll not stop it, however one should consider doing
```

Comment:
this really concerns me.  I spend most of my time with my personal virtualenv activated which is writable.  I don't want some random thing I'm developing on to muck with that.  This feature allows them to do that.

Reply:
hence why not allowing by default but giving the option if they are explicit about it (aka commands only). Oliver was in favour of this a while back, not sure if he changed his mind.

Reply:
I think that stops being ok because people fork and develop on other people's software.  granted they can already do some crazy stuff with `tox.ini` today, but this feature makes it way easier to do so on accident and without warning.

Reply:
Yeah I definitely have such worries hence my comments on the issue and disallowing deps and enforcing skip install true... It's non trivial to do changes by intent or accident. Requiring explicit install in command felt a decent middle ground. 🤔

Reply:
I went to check out this branch to play with this, and it seems even when using `host_env` that a tox environment gets created.

Even if the feature worked (?) this feature "optimizes" 3 seconds away on first run and then no optimization on second+ runs.  I don't think that's worth enabling an easy to abuse feature.

```ini
[testenv:old]
skip_install = true
deps =
commands =
    python -c 'print("hello world")'
```

```console
$ time tox -e old
old create: /tmp/x/x/.tox/old
old runtests: commands[0] | python -c print("hello world")
hello world
___________________________________ summary ____________________________________
  old: commands succeeded
  congratulations :)

real	0m3.454s
user	0m3.119s
sys	0m0.227s

$ time tox -e old
old runtests: commands[0] | python -c print("hello world")
hello world
___________________________________ summary ____________________________________
  old: commands succeeded
  congratulations :)

real	0m0.107s
user	0m0.095s
sys	0m0.012s
```

Reply:
it does work, see the test added with this pr,  your pr doesn't enable the host tox env....

Reply:
lol, so I did `git checkout 1181` and instead of checking out your branch it checked out a sha with that prefix 🤦‍♂️ 

Here's more accurate timing differences:

```console
$ time tox -e new
new installed: filelock==3.0.10,pluggy==0.9.0,py==1.8.0,six==1.12.0,toml==0.10.0,tox==3.7.1.dev24+g455e4de,virtualenv==16.4.3
new run-test-pre: PYTHONHASHSEED='2687712007'
new runtests: commands[0] | python -c 'print("hello world")'
hello world
___________________________________ summary ____________________________________
  new: commands succeeded
  congratulations :)

real	0m0.557s
user	0m0.516s
sys	0m0.040s
$ time tox -e old
old create: /tmp/x/x/.tox/old
old run-test-pre: PYTHONHASHSEED='2024296911'
old runtests: commands[0] | python -c 'print("hello world")'
hello world
___________________________________ summary ____________________________________
  old: commands succeeded
  congratulations :)

real	0m3.868s
user	0m3.528s
sys	0m0.244s
$ time tox -e old
old run-test-pre: PYTHONHASHSEED='1190934059'
old runtests: commands[0] | python -c 'print("hello world")'
hello world
___________________________________ summary ____________________________________
  old: commands succeeded
  congratulations :)

real	0m0.535s
user	0m0.503s
sys	0m0.032s
```

I wonder what has regressed 400ms of startup time since the old version I was running as well 🤔 

with the current startup time, the savings is even less pronounced

Reply:
it's much more pronounced on Windows to be fair, for the unit test on my machine its a 20 second difference...

Reply:
that's worse yeah, but only for the first run

I think you can also avoid that by setting `envdir` to some other tox env right?

Reply:
well sometimes only at first run is not a good enough excuse,  it could be fast straight from out of box  😃🤔

Reply:
Hi, sorry that I didn't chip in here. It completely went under my Radar. It's too late, but I agree with your decision anyway. 

I think nowadays that having the need to do that is an "infrastructure smell", meaning that it shows that there might be a deeper problem with the way how the project is structured or there are some huge problems regarding the ease of making dependencies available.

Something like this would have no business in any OSS project I think and I don't know if any project that could profit for that. 

As this might tamper with an environment that should not be tampered with this is definitely "dangerous", so not advisable in this form. 

But the situation is usually different behind corporate firewalls there it might be a quick fix in some existing projects that have been built with different standards and assumption. Making the developer workflow more pleasant with hacks like this one, might help tidying them over to a cleaner, more reproducible build, test and deploy workflow. Then those hacks can be removed again.

That's why I think It's better not in core. Not many people ask for it (and I really think they shouldn't). But having this in a plugin with a caveat might be helpful in cases like described above.

I started a plugin yesterdy that just scratched an itch without knowing about this. I will have a look what you did here and see if I can turn that plugin into something useful that we could refer to for situations like this.